### PR TITLE
Modernised. Untested.

### PR DESCRIPTION
**Pull request by simonccarter**
Untested! I don't have Spotify api keys. Please test this, especially promises.

Run locally on node version 6.9.1.

Changes include:

Use arrow functions
Use pure functions
Replace var with let/const
Replace initial if/else with just single if
Move what looks like unnecessary youtube initialisation outside of youtube function to global scope.
Replace for loop with forEach
Replace sys with util
Global ternary statement rather than unnecessary local if/else for cmd string init

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,82 +1,66 @@
...
+    let url = createYouTubeUrl(createYouTubeId(res))
+    console.log('Youtube url: '+url);
+    dl(url);
+  }
+}
 
-var clientId = config.spotify.clientid,
-    clientSecret = config.spotify.clientsecret;
+const dlCb = (err, stdout, stderr) => {
+  if(error) console.log('exec error: '+error);
```
Comment (adhorrig) [inline src/index.js]: ```Error``` is not defined, should be using ```err``` here.

File: src/index.js [hunk 2]
```diff
@@ -1,82 +1,66 @@
...
+  else      util.print('stdout: '+stdout);
```
Comment (adhorrig) [inline src/index.js]: Could this whitespace be reduced?

**Review by adhorrig — CHANGES_REQUESTED** [review]
Mostly looks great, I noticed two small things though!

**Comment by simonccarter** [comment]
Pushed an update to include your requested changes. If you want to test this before merge, you can always pull from my fork :)

File: src/index.js [hunk 3]
```diff
@@ -37,8 +37,8 @@ const ytCb = (err, res) => {
 }
 
 const dlCb = (err, stdout, stderr) => {
-  if(error) console.log('exec error: '+error);
-  else      util.print('stdout: '+stdout);
+  if(er) console.log('exec error: '+error);
```
Comment (adhorrig) [inline src/index.js]: Forgot one 'r' :), also the ```error``` in the print statement should also be ```err``` too

**Comment by adhorrig** [comment]
I tested when reviewing and that was the only issue, I added a comment to the latest commit. After that it should be good to be merged 😃

**Comment by simonccarter** [comment]
And again. Rebased.

**Review by adhorrig — APPROVED** [review]

**Comment by adhorrig** [comment]
Perfect 👍 Thanks a bunch!

**Pull request merged**